forte/data/data_store.py

```diff
@@ -224,8 +245,15 @@ def _new_link(
 
         tid: int = self._new_tid()
         entry: List[Any]
+
+        self._add_new_type(type_name)
```

Comment:
You should get the type attribute content from this call already, don't need to call again.